src/util/transport.rs

```diff
@@ -61,7 +61,7 @@ pub enum NotifyError<T> {
     Io(io::Error),
 }
 
-pub trait Sender<T>: Clone {
+pub trait Sender<T: Sized>: Clone + Sized {
```

Comment:
Why `Sized`?

Reply:
SizeCheckObserver contains a `RetryableSendCh<Msg, C> where C: Sender<Msg>`, and it will be made into a trait object, so `Sized` for `Sender<T>`, otherwise it cant be a trait object.